### ISSUE TITLE
Throttle -> Ele Mix not usable with User Curve

### ISSUE THREAD

**Issue by Hartinger** [FrSkyRC/ETHOS-Feedback-Community#2579]
In ETHOS 1.4.8 there is now a hidden curve in the THR -> ELE Mix, transforming the throttle stick range (-100% ... 100%) into a range of 0 ... 100%, which is fine and as required. However, in reality you may want to use your own curve for this mix, and this is currently not really possible, as shown below. (I have generated the following screenshots with the Tandem X20S simulator.)

If you define your own curve over the complete stick range, e.g.:

![thr_ele_1](https://user-images.githubusercontent.com/22352021/231736599-7dccf9f9-bc55-41f5-a227-4f630564e5ab.png)

then only the right half of the curve is used and squeezed into the lower half of the throttle stick range. In particular, this means that at the idle position of the throttle stick the mixer output is not 0! (I assume the hidden curve is executed first.)

![thr_ele_2](https://user-images.githubusercontent.com/22352021/231737364-861c85d1-fa0d-4b21-8b40-f1fc09509d52.png)

If the own curve is then modified to only use the 0 ... 100% range generated by the hidden curve, e.g.:

![thr_ele_3](https://user-images.githubusercontent.com/22352021/231737866-e1521e4c-8b57-4db5-bdd2-2822cae8a396.png)

then the own curve is still squeezed into the lower half of the throttle stick range; in the upper half of the stick range the mixer output remains constant.

![thr_ele_4](https://user-images.githubusercontent.com/22352021/231738904-f04e0d6f-9138-423e-8dab-1498529276f4.png)

In summary, the THR -> ELE Mix is not usable if one needs a curve different to the hidden default curve.

In addition, all mixers with Throttle as source should really take into account the Throttle Cut-Off setting, which they don't at the moment. (As a workaround, you can switch off the mix based on the corresponding system event.)

**Comment by bsongis-frsky** [comment]
It's not an hidden curve but an offset applied to the input, so that the input is moved inside the range [0% : +100%]
But the offset should be applied **after** the custom curve, because the curve really needs the whole range [-100% : +100%]

**Comment by bsongis-frsky** [comment]
Here is the curve you will need to get the result you expect:

![image](https://user-images.githubusercontent.com/50834778/231991931-4e2f1ec3-36a8-43ba-9e50-a882317f2858.png)

![image](https://user-images.githubusercontent.com/50834778/231992111-df5dbed5-8fa8-455e-b4c8-38f43d662fef.png)

**Comment by hrenz** [comment]
and the same old problem  with die Butterfly mixer with die "hidden" curve or Offset too

Let the user decide what he wants to do how, with values and curves

**Comment by Hartinger** [comment]
Hi,

I think that the best and most user friendly implementation would be as follows.

If no curve is assigned to the mixer, it should behave exactly as it does now, i.e. the throttle stick range is transformed into a mixer output in the range from 0 (idle position) to w% (full throttle), where w respresents the weight.

If the user has assigned a curve (no matter which one), then any internal offset/curve should be disregarded, and the mixer output should be solely driven by the assigned curve. This allows the user to define the curve in such a way that it represents the desired mixer output as a function of the mixer input (-100 ... +100%). This is far more intuitive than if one has to consider some mixer internal transformation for the construction of the curve.

The same principle should apply to any othe mixer that has an inherent transformation of the input range. The Butterfly mixer in particular is another example where this logic should be implemented.

**Comment by bsongis-frsky** [comment]
It makes sense to me

**Comment by bsongis-frsky** [comment]
It will have impact on all mixes which behave the same:
- Thr => Ele
- Thr => Rud
- Ail => Rud
- Rud => Ail
- Ele => Ail
- Rud => Ele

**Comment by bsongis-frsky** [comment]
![image](https://user-images.githubusercontent.com/50834778/232104780-11de1144-0f8b-465b-a5f8-1ebaa3a99300.png)

![image](https://user-images.githubusercontent.com/50834778/232104836-0b929693-5e4b-41d7-9045-7e356de7ef23.png)

**Comment by Hartinger** [comment]
The last four mixes in your list

Ail => Rud
Rud => Ail
Ele => Ail
Rud => Ele

do not fall into this category. They are fully symmetrical mixers with no internal transformation (other than the application of the weight). So any assigned curve would already now exclusively determine the mixer characteristic, no need for any change.

**Comment by bsongis-frsky** [comment]
True. There should be no impact on them, but they just need some tests because they are part of the same CombiMix category which has been modified.

**Comment by Hartinger** [comment]
The Butterfly mixer does fall into this category, i.e. a user assigned curve should **replace** the default processing. However, we need to be aware that there would be an impact on existing setups.

**Comment by bsongis-frsky** [comment]
Not exactly. The butterfly is not the same kind of mix, its input is configurable. It needs some love too :)

**Comment by bsongis-frsky** [comment]
![image](https://user-images.githubusercontent.com/50834778/232113819-ff96bd5b-5b95-4d37-883b-0695b20bd88b.png)

**Comment by bsongis-frsky** [comment]
Butterfly enabled when the Throttle stick is > 0:

![image](https://user-images.githubusercontent.com/50834778/232114764-7ae35be2-27a5-4813-917c-2b189dd1ec78.png)

**Comment by bsongis-frsky** [comment]
![image](https://user-images.githubusercontent.com/50834778/232114811-db87b2cf-90a2-4b0a-bedd-14014a1e9c24.png)

**Comment by bsongis-frsky** [comment]
This one is merged to the 1.5.0 nightlies, it would be cool if someone could give it a test during the week-end and check if I didn't introduce any regression!

**Comment by Hartinger** [comment]
As this change will have an impact on existing setups, can I suggest to put a clear hint to that effect into the relase notes.

**Comment by bsongis-frsky** [comment]
> As this change will have an impact on existing setups, can I suggest to put a clear hint to that effect into the relase notes.

As usual!

**Comment by Hartinger** [comment]
BTW: I wanted to test the nightly version of the X20S simulator and found that a vertical stick movement is only possible in the negative direction.

**Comment by MiRe-CZ** [comment]
Well, I thought the problem with sticks is on my side, nobody else reported it and it is not fresh issue, it was in couple of previous nightly X20S simulator versions.
So I can confirm - both "8" and "Ctrl 8" are interpreted as FS6, moreover both "7" and "Ctrl 7" are interpreted as FS5
